src/main/java/org/zalando/nakadi/service/subscription/zk/NewZkSubscriptionClient.java

```diff
@@ -131,20 +139,49 @@ private Topology parseTopology(final byte[] data) {
     }
 
 
-    @Override
-    public Partition[] listPartitions() throws NakadiRuntimeException, SubscriptionNotInitializedException {
-        return readTopology().getPartitions();
-    }
-
     protected String getOffsetPath(final EventTypePartition etp) {
         return getSubscriptionPath("/offsets/" + etp.getEventType() + "/" + etp.getPartition());
     }
 
     @Override
-    public SubscriptionCursorWithoutToken getOffset(final EventTypePartition key) throws NakadiRuntimeException {
+    public Map<EventTypePartition, SubscriptionCursorWithoutToken> getOffsets(final EventTypePartition[] keys)
+            throws NakadiRuntimeException {
+        final Map<EventTypePartition, SubscriptionCursorWithoutToken> result = new HashMap<>();
         try {
-            final String offset = new String(getCurator().getData().forPath(getOffsetPath(key)), UTF_8);
-            return new SubscriptionCursorWithoutToken(key.getEventType(), key.getPartition(), offset);
+            final CountDownLatch latch = new CountDownLatch(keys.length);
```

Comment:
It looks like I see this part of code twice in this PR, do you think we could move it out the method?